pkg/streamingpromql/planning.go

```diff
@@ -126,7 +126,7 @@ func (p *QueryPlanner) NewQueryPlan(ctx context.Context, qs string, timeRange ty
 	}
 
 	expr, err = p.runASTStage("Pre-processing", observer, func() (parser.Expr, error) {
-		return promql.PreprocessExpr(expr, timestamp.Time(timeRange.StartT), timestamp.Time(timeRange.EndT))
+		return promql.PreprocessExpr(expr, timestamp.Time(timeRange.StartT), timestamp.Time(timeRange.EndT), time.Duration(timeRange.IntervalMilliseconds)*time.Millisecond)
```

Comment:
For instant queries, interval will be 1. Prometheus expects `step` to be 0. I'm not sure if this will cause problems, but we might want to translate here:

```go
interval := timeRange.IntervalMilliseconds)*time.Millisecond
if timeRange.isInstant { interval = 0 }
```

Reply:
Good catch, fixed in c02accd.